### PR TITLE
[Keyboard] Add Dactyl 6x5 68 keys

### PR DESCRIPTION
**Pull request by kpagratis**
Dactyl Manuform 5x6 with 4 button bottom row and 6 button thumb clusters totaling 68 keys. 
## Checklist

- [ x] My code follows the code style of this project: [**C**](https://docs.qmk.fm/#/coding_conventions_c), [**Python**](https://docs.qmk.fm/#/coding_conventions_python)
- [ x] I have read the [**PR Checklist** document](https://docs.qmk.fm/#/pr_checklist) and have made the appropriate changes.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [ x] I have read the [**CONTRIBUTING** document](https://docs.qmk.fm/#/contributing).
- [ ] I have added tests to cover my changes.
- [ ] I have tested the changes and verified that they work and don't break anything (as well as I can manage).

### CODE REVIEW

File: keyboards/handwired/dactyl_manuform/5x6_68/5x6_68.c [hunk 1]
```diff
@@ -0,0 +1 @@
+#include "5x6_68.h"
```
Comment (drashna) [inline keyboards/handwired/dactyl_manuform/5x6_68/5x6_68.c:1]: Could you add a GPL2+ compatible license header here?  All of the source files need to have a license explicitly listed.

For instance:
```c
// Copyright 2021 Your Name (@yourgithub)
// SPDX-License-Identifier: GPL-2.0-or-later
```

File: keyboards/handwired/dactyl_manuform/5x6_68/5x6_68.h [hunk 2]
```diff
@@ -0,0 +1,63 @@
+#pragma once
```
Comment (drashna) [inline keyboards/handwired/dactyl_manuform/5x6_68/5x6_68.h:1]: Could you add a GPL2+ compatible license header here?  All of the source files need to have a license explicitly listed.

For instance:
```c
// Copyright 2021 Your Name (@yourgithub)
// SPDX-License-Identifier: GPL-2.0-or-later
```

File: keyboards/handwired/dactyl_manuform/5x6_68/keymaps/default/keymap.c [hunk 3]
```diff
@@ -0,0 +1,54 @@
+#include QMK_KEYBOARD_H
```
Comment (drashna) [inline keyboards/handwired/dactyl_manuform/5x6_68/keymaps/default/keymap.c:1]: Could you add a GPL2+ compatible license header here?  All of the source files need to have a license explicitly listed.

For instance:
```c
// Copyright 2021 Your Name (@yourgithub)
// SPDX-License-Identifier: GPL-2.0-or-later
```

File: keyboards/handwired/dactyl_manuform/5x6_68/keymaps/default/keymap.c [hunk 4]
```diff
@@ -0,0 +1,54 @@
...
+
+#define _QWERTY 0
+#define _LAYER2 1
```
Comment (drashna) [inline keyboards/handwired/dactyl_manuform/5x6_68/keymaps/default/keymap.c:4]:
```suggestion
enum custom_layers {
    _QWERTY,
    _LAYER2,
};
```

**Review by drashna — CHANGES_REQUESTED** [review]

**Comment by kpagratis** [comment]
I'll get on these changes. I didn't get an email or anything

**Comment by kpagratis** [comment]
Apologies for the delay. I've updated the pr

File: keyboards/handwired/dactyl_manuform/5x6_68/config.h [hunk 5]
```diff
@@ -0,0 +1,46 @@
...
+
+// wiring of each half
+#define MATRIX_COL_PINS { D4, C6, D7, E6, B4, B5 }
+#define MATRIX_ROW_PINS { F6, F7, B1, B3, B2, B6 }
+
+#define DIODE_DIRECTION ROW2COL
+#define USE_SERIAL
+#define SOFT_SERIAL_PIN D0
+#define SPLIT_USB_DETECT
+
```
Comment (waffle87) [inline keyboards/handwired/dactyl_manuform/5x6_68/config.h:36]:
```suggestion
// wiring of each half
#define MATRIX_COL_PINS { D4, C6, D7, E6, B4, B5 }
#define MATRIX_ROW_PINS { F6, F7, B1, B3, B2, B6 }

#define DIODE_DIRECTION ROW2COL
#define SPLIT_USB_DETECT

```

File: keyboards/handwired/dactyl_manuform/5x6_68/keymaps/default/keymap.c [hunk 6]
```diff
@@ -0,0 +1,60 @@
...
+#define CTL_S   LCTL_T(KC_S)
+#define ALT_D   LALT_T(KC_D)
+#define ALT_K   RALT_T(KC_K)
+#define CTL_L   RCTL_T(KC_L)
+#define SFT_SC  RSFT_T(KC_SCLN)
+#define IMPORTS LALT(LCTL(KC_O))    //Intellij Organize Imports
+#define LINES   LALT(LGUI(KC_L))    //Intellij Reformat Code
+#define USAGES  LALT(KC_F7)         //Intelllj Find Usages
+#define RENAME  LSFT(KC_F6)         //Intellij Refactor Rename
+#define xxxxxxx KC_NO
```
Comment (waffle87) [inline keyboards/handwired/dactyl_manuform/5x6_68/keymaps/default/keymap.c:23]: This alias [already exists](https://github.com/qmk/qmk_firmware/blob/master/quantum/keycodes.h#L719) and should be used instead.
```suggestion
```

File: keyboards/handwired/dactyl_manuform/5x6_68/rules.mk [hunk 7]
```diff
@@ -0,0 +1,19 @@
+# MCU name
+MCU = atmega32u4
+
+# Bootloader selection
+BOOTLOADER = caterina
+
```
Comment (waffle87) [inline keyboards/handwired/dactyl_manuform/5x6_68/rules.mk:6]:
```suggestion
```

File: keyboards/handwired/dactyl_manuform/5x6_68/5x6_68.h [hunk 8]
```diff
@@ -0,0 +1,66 @@
...
+    { L50, L51, L52, L53, L54, L55 }, \
+\
+    { R00, R01, R02, R03, R04, R05 }, \
+    { R10, R11, R12, R13, R14, R15 }, \
+    { R20, R21, R22, R23, R24, R25 }, \
+    { R30, R31, R32, R33, R34, R35 }, \
+    { XXX, XXX, R42, R43, R44, R45 }, \
+    { R50, R51, R52, R53, R54, R55 } \
+}
+*/
```
Comment (waffle87) [inline keyboards/handwired/dactyl_manuform/5x6_68/5x6_68.h:66]: This file should be deleted (assuming the `matrix` data in info.json file is correct)

File: keyboards/handwired/dactyl_manuform/5x6_68/5x6_68.c [hunk 9]
```diff
@@ -0,0 +1,4 @@
+// Copyright 2023 Kostas Pagratis (@kpagratis)
```
Comment (waffle87) [inline keyboards/handwired/dactyl_manuform/5x6_68/5x6_68.c:1]: This file should be deleted (proven my above comment is true)

File: keyboards/handwired/dactyl_manuform/5x6_68/config.h (hunk 5 above)
Comment (kpagratis) [reply keyboards/handwired/dactyl_manuform/5x6_68/config.h:36]: Wouldn't removing MATRIX_ROWS and MATRIX_COLS break keymap.c?

Comment (kpagratis) [reply keyboards/handwired/dactyl_manuform/5x6_68/config.h:36]: nevermind. using the info.json only removes this dependency

**Comment by kpagratis** [comment]
Thank you for taking the time with this. I have your changes completed locally.  I'll flash the keyboard tomorrow and update the PR

**Comment by kpagratis** [comment]
these changes borked the layout. I need to spend some time understanding the relationship between info.json and keymap.c

**Comment by kpagratis** [comment]
Ok this is ready for review again. Thanks again

**Comment by drashna** [comment]
It looks like you have changes to the submodules here.  These need to be reverted before the PR can be approved. 

Rebasing sometimes will fix this. Otherwise, you'd want to do something like this to fix them:
```sh
git checkout master
git submodule foreach 'git log -n1 --oneline'
# remember the first abcdef1234 on the submodules that have an issue
git checkout @@your-branch@@
# for each of the problematic repo's:
cd lib/@@name@@
git checkout abcdef1234
```

**Comment by kpagratis** [comment]
Oh weird! OK I'll fix

**Comment by kpagratis** [comment]
lol ok I got there eventually. I've fixed the submodules

**Comment by kpagratis** [comment]
Is anything else needed?

**Review by waffle87 — APPROVED** [review]

File: keyboards/handwired/dactyl_manuform/5x6_68/config.h [hunk 10]
```diff
@@ -0,0 +1,8 @@
+// Copyright 2023 Kostas Pagratis (@kpagratis)
+// SPDX-License-Identifier: GPL-2.0-or-later
+
+#pragma once
+
+#include "config_common.h"
+
+#define SPLIT_USB_DETECT
```
Comment (drashna) [inline keyboards/handwired/dactyl_manuform/5x6_68/config.h:8]: This is something that should probably go into the keymap config.
```suggestion
```

File: keyboards/handwired/dactyl_manuform/5x6_68/keymaps/default/config.h [hunk 11]
```diff
@@ -0,0 +1,9 @@
+// Copyright 2023 Kostas Pagratis (@kpagratis)
+// SPDX-License-Identifier: GPL-2.0-or-later
+
+#pragma once
+
+
+#define MASTER_LEFT
```
Comment (drashna) [inline keyboards/handwired/dactyl_manuform/5x6_68/keymaps/default/config.h:7]: Master left isn't needed, as this is the default behavior if no other handedness is set. 

Also, split usb detect would probably be better here. 
```suggestion
#define SPLIT_USB_DETECT
```

File: keyboards/handwired/dactyl_manuform/5x6_68/keymaps/default/keymap.c [hunk 12]
```diff
@@ -0,0 +1,59 @@
...
+#define ALT_K   RALT_T(KC_K)
+#define CTL_L   RCTL_T(KC_L)
+#define SFT_SC  RSFT_T(KC_SCLN)
+#define IMPORTS LALT(LCTL(KC_O))    //Intellij Organize Imports
+#define LINES   LALT(LGUI(KC_L))    //Intellij Reformat Code
+#define USAGES  LALT(KC_F7)         //Intelllj Find Usages
+#define RENAME  LSFT(KC_F6)         //Intellij Refactor Rename
+
+const uint16_t PROGMEM keymaps[][MATRIX_ROWS][MATRIX_COLS] = {
+    [_QWERTY] = LAYOUT(
```
Comment (drashna) [inline keyboards/handwired/dactyl_manuform/5x6_68/keymaps/default/keymap.c:25]: This should be side by side, not stacked. (and the info.json probably needs to be updated to reflect that, as well.)

**Review by drashna — CHANGES_REQUESTED** [review]

File: keyboards/handwired/dactyl_manuform/5x6_68/keymaps/default/keymap.c (hunk 12 above)
Comment (kpagratis) [reply keyboards/handwired/dactyl_manuform/5x6_68/keymaps/default/keymap.c:25]: I'm trying to wrap my head around how to make the halves side by side in the info.json and I can't wrap my brain around how that definition translates to the pinouts on the MCs

**Comment by kpagratis** [comment]
I think all the comments have been addressed

**Review by drashna — APPROVED** [review]

File: keyboards/handwired/dactyl_manuform/5x6_68/config.h [hunk 13]
```diff
@@ -0,0 +1,7 @@
+// Copyright 2023 Kostas Pagratis (@kpagratis)
```
Comment (waffle87) [inline keyboards/handwired/dactyl_manuform/5x6_68/config.h:1]: This file should just be removed.

Comment (kpagratis) [reply keyboards/handwired/dactyl_manuform/5x6_68/config.h]: done

**Pull request merged**